crates/router/src/connector/nmi/transformers.rs

```diff
@@ -575,6 +611,35 @@ impl TryFrom<&domain::PaymentMethodData> for PaymentMethod {
     }
 }
 
+impl TryFrom<(&domain::payments::Card, &types::PaymentsAuthorizeData)> for PaymentMethod {
+    type Error = Error;
+    fn try_from(
+        val: (&domain::payments::Card, &types::PaymentsAuthorizeData),
+    ) -> Result<Self, Self::Error> {
+        let (card_data, item) = val;
+        let auth_data = &item.get_authentication_data()?;
+        let ccexp = utils::CardData::get_card_expiry_month_year_2_digit_with_delimiter(
+            card_data,
+            "".to_string(),
+        )?;
+
+        let card_3ds_details = CardThreeDsData {
+            ccnumber: card_data.card_number.clone(),
+            ccexp,
+            cvv: card_data.card_cvc.clone(),
+            email: item.email.clone(),
+            cavv: Some(auth_data.cavv.clone()),
+            eci: auth_data.eci.clone(),
+            xid: None,
```

Comment:
Please populate xid with threeds_server_transaction_id and three_ds_version with message_version here